### PR TITLE
ESBJAVA-5060 Support HTTP DELETE with content body

### PR DESCRIPTION
**Pull request by milindaperera**
This Fix will provide support for HTTP DELETE Request with entity body.

JIRA: https://wso2.org/jira/browse/ESBJAVA-5060

### CODE REVIEW

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java [hunk 1]
```diff
@@ -384,7 +385,8 @@ private void sendRequestContent(final MessageContext msgContext) throws AxisFaul
                 } else {
                     //if HTTP MEHOD = GET we need to write down the HEADER information to the wire and need
                     //to ignore any entity enclosed methods available.
-                    if (("GET").equals(msgContext.getProperty(Constants.Configuration.HTTP_METHOD)) || ("DELETE").equals(msgContext.getProperty(Constants.Configuration.HTTP_METHOD))) {
+                    if (("GET").equals(msgContext.getProperty(Constants.Configuration.HTTP_METHOD)) ||
```
Comment (madhawa-gunasekara) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java]: Shall we use HttpMethod.GET here ?

Comment (milindaperera) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java]: thanks, will do

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java [hunk 2]
```diff
@@ -111,8 +111,8 @@ public static TargetRequest create(MessageContext msgContext,
             }
 
 			String cType = getContentType(msgContext, configuration.isPreserveHttpHeader(HTTP.CONTENT_TYPE));
-			if (cType != null && (!httpMethod.equals("GET") && !httpMethod.equals("DELETE"))) {
-				String messageType = (String) msgContext.getProperty("messageType");
+			if (cType != null && (!httpMethod.equals("GET") && !RelayUtils.isDeleteRequestWithoutPayload(msgContext))) {
```
Comment (erandacr) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java]: Line length can be corrected

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ServerWorker.java [hunk 3]
```diff
@@ -684,8 +684,7 @@ public void processHttpRequestUri(MessageContext msgContext, String method){
         msgContext.setTo(new EndpointReference(restUrlPostfix));
         msgContext.setProperty(PassThroughConstants.REST_URL_POSTFIX, restUrlPostfix);
 
-        if (HttpMethod.GET.equals(method) || HttpMethod.DELETE.equals(method)  ||  HttpMethod.HEAD.equals(method)||
-                                                                                             "OPTIONS".equals(method)) {
+        if (HttpMethod.GET.equals(method) ||  HttpMethod.HEAD.equals(method)|| "OPTIONS".equals(method)) {
```
Comment (erandacr) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ServerWorker.java]: Don't we have a HttpMethod.OPTIONS constant

Comment (milindaperera) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ServerWorker.java]: Options not available in javax.ws.rs.HttpMethod

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java (hunk 1 above)
Comment (milindaperera) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java]: Fixed in https://github.com/wso2/wso2-synapse/pull/801/commits/b6c89cf2bdf920a6e2a44ab6b3f1fbc2fabcf145

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java (hunk 2 above)
Comment (milindaperera) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java]: Fixed in https://github.com/wso2/wso2-synapse/pull/801/commits/b6c89cf2bdf920a6e2a44ab6b3f1fbc2fabcf145

**Comment by milindaperera** [comment]
Closing this PR to fix commits.
Thanks for reviewing

**Pull request closed without merging**